### PR TITLE
Release checklist: replace projects with dotcom and update the items to reflect recent practises

### PR DESCRIPTION
**Pull request by siddharthkp**
1. Replaced projects with github/github
2. Added link to finding automated integration pull request
3. Update the checklist to reflect our current practises and decisions

### CODE REVIEW

**Review by joshblack — APPROVED** [review]

File: .github/release_template.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ Please ensure these items are checked before merging.
 
 - [ ] All CI checks pass
 - [ ] Docs and Storybook open in a browser
-- [ ] Successful integration test with GitHub Projects as a primary consumer of Primer React
+- [ ] Successful integration test with github/github as a primary consumer of Primer React ([automated pull request](https://github.com/github/github/pulls?q=is%3Apr+is%3Aopen+Integration+tests+for+%40primer%2Freact+label%3Adependencies))
   - [ ] Install the Release Candidate
   - [ ] Verify no new build errors appear
   - [ ] Verify no new linting errors appear
```
Comment (broccolinisoup) [inline .github/release_template.md:12]: I wonder if we should extend this bullet to mention the path forward using the integration PR as well along with the manual installation checks below? 

Like;

Successful integration test with github/github as a primary consumer of Primer React 
- Manually installing the `@next`
  - [ ]  Install the Release Candidate
  - [ ]  Verify no new build errors appear
  - [ ]  Verify no new linting errors appear
  - [ ]  Verify no new browser console errors appear
  - [ ]  Verify unit tests and E2E tests pass
  - [ ]  Manually test critical paths (Tip: Use the provided tests project boards)
  - [ ]  Manually test release-specific bugfixes and/or features work as described
- Utilizing the  ([automated integration pull request](https://github.com/github/github/pulls?q=is%3Apr+is%3Aopen+Integration+tests+for+%40primer%2Freact+label%3Adependencies))
  - [ ]  Manually test critical paths in the review-lab
  - [ ]  Manually test release-specific bugfixes and/or features work as described in the review-lab

>(Tip: Use the provided tests project boards)

Also I don't think we need this tip anymore?

Comment (siddharthkp) [reply .github/release_template.md:12]: We can probably condense all the manual steps to just verifying checks on the PR?

```diff
- [ ] Install the Release Candidate
- [ ] Verify no new build errors appear
- [ ] Verify no new linting errors appear
- [ ]  Verify no new browser console errors appear
- [ ] Verify unit tests and E2E tests pass

+ [ ] Verify all checks have passed on automated pull request
 ```

Comment (siddharthkp) [reply .github/release_template.md:12]: Do we want to take that up during release or leave it for the upgrade PR?

```
- [ ] Manually test critical paths in the review-lab
- [ ] Manually test release-specific bugfixes and/or features work as described in the review-lab
```

Comment (broccolinisoup) [reply .github/release_template.md:12]: I am not proud of my very very late response here, so I apologise! 😬 

>We can probably condense most of the manual steps to just verifying checks on the PR?

I like that!

>Do we want to these up during release or leave it for the upgrade PR?

My humble opinion on this is that we should keep it on the release PR. I am imagining the below steps for release conductor:
 
- We would want to see the checks are passing on the integration PR
- Approve the integration PR and deploy it to the review-lab
- And do the manual checks (the 2 checklist items above)
- Cut the release
- Upgrade the version at dotcom

Is this aligning with what you were thinking @siddharthkp ?

**Review by lesliecdubs — APPROVED** [review]

File: .github/release_template.md [hunk 2]
```diff
@@ -4,29 +4,30 @@ Please ensure these items are checked before merging.
...
+  - [ ] Verify all CI checks have passed on automated pull request
+  - [ ] Manually test critical paths in the review-lab
+  - [ ] Manually test release-specific bugfixes and/or features work as described in the review-lab
+
 ### 🤔 Sanity test
 
-- [ ] All bugfixes in this release have resolved their corrosponding issues
-- [ ] All new features in this release have been tested and verified as compatible with GitHub Projects
+- [ ] All bugfixes in this release have resolved their corresponding issues
+- [ ] Reverted PRs' issues have been re-opened and commented on with a link to the reverted PR
```
Comment (joshblack) [inline .github/release_template.md:23]: Small suggestion!

```suggestion
- [ ] The issues for reverted PRs have been re-opened and commented on with a link to the reverted PR
```

**Review by joshblack — APPROVED** [review]

**Pull request merged**